src/bin/ion/commands/generate/utils.rs

```diff
@@ -50,6 +50,10 @@ pub trait Language {
     ///     In Rust, Template::Struct -> "struct"
     ///     In Java, Template::Struct -> "class"
     fn template_name(template: &Template) -> String;
+
+    fn namespace_separator() -> String;
+
+    fn add_type_to_namespace(is_nested_type: bool, type_name: &String, namespace: &mut Vec<String>);
```

Comment:
Why is `is_nested_type` necessary for this? Please include doc comments that explain the contract for this function.